packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/ModelTestCaseUtil.java

```diff
@@ -105,7 +105,8 @@ public static Object jsonFromType(int depth, IType type) {
         } else if (type.asNullable() == ClassType.DOUBLE) {
             return RANDOM.nextDouble() * 100;
         } else if (type.asNullable() == ClassType.BOOLEAN) {
-            return RANDOM.nextBoolean();
+            // always returns false for boolean, for HEAD operation
+            return false;
```

Comment:
Why it is always false?

If it is only for HEAD, we seems only need to do special handling in FluentMethodMockTestTemplate, even hardcode it as `assertTrue` or `assertFalse` on response?

Reply:
Yes, we could hardcode the assertion in mock test case. Let me update the code.

Reply:
updated